### DISCUSSION TITLE
Image component - add you'r own loader

### DISCUSSION THREAD

**Discussion by pontusab** [vercel/next.js#18450] [Ideas]
**Edit:** Answered below https://github.com/vercel/next.js/discussions/18450#discussioncomment-1040243

Could not find if this is possible, but we have our own resize service that we would like to use in the Image component, would it be possible to add our own custom Loader?

**Comment by matamatanot** [comment]
I hope for that too.

https://github.com/vercel/next.js/issues/18122#issuecomment-717770881

**Comment by timneutkens** [comment]
We didn't add it initially as it complicated the feature a lot, we'll add it at a later point in time 👍

**Reply by pontusab** [reply]
Thanks! Sounds great!

**Reply by sandcastle** [reply]
@timneutkens - how can we help out with getting this added in? We run our own image optimisation service. This is a really big blocker not having custom loaders for using the `next/image` component.

**Reply by timneutkens** [reply]
It's being worked on: https://github.com/vercel/next.js/pull/19325

**Comment by mlibk** [comment]
Am I correct in thinking that, while there isn't an official way to add a custom loader (yet), I could simply pick one of the in-built ones e.g. 'imgix' and provide path to my custom image service that mimicks the Imgix API?

**Reply by pontusab** [reply]
True! That's what we ended up with!

**Comment by matamatanot** [comment]
There are currently several Loader additions PRs out.

#18314
#18391
#18495
#18561

It's hard to decide what to support and what not to support.

**Comment by leerob** [comment]
This is now possible! https://nextjs.org/docs/basic-features/image-optimization#loader
